tests/test_no_numpy_builtin_type_aliases.py

```diff
@@ -0,0 +1,88 @@
+import ast
+import os
+
+
+def iterate_python_scripts(directory):
+    for root, _, files in os.walk(directory):
+        for f in files:
+            if f.endswith(".py"):
+                yield os.path.join(root, f)
+
+
+def read_file(path):
+    with open(path) as f:
+        return f.read()
+
+
+def is_builtin_type_alias(attr_node):
+    """
+    Returns True if the passed-in attribute node represents a NumPy's alias for a built-in type
+    (e.g. np.int)
+    """
+    return (
+        isinstance(attr_node.value, ast.Name)
+        and attr_node.value.id in ["numpy", "np"]
+        and attr_node.attr
+        in ["bool", "complex", "float", "int", "long", "object", "str", "unicode"]
+    )
+
+
+class NumpyBuiltinAliasFinder(ast.NodeVisitor):
+    def __init__(self):
+        super().__init__()
+        self.found = []
+        self.path = None
+
+    def visit_Attribute(self, node):
+        if is_builtin_type_alias(node):
+            self.found.append((self.path, node))
+        self.generic_visit(node)
+
+    def find(self, path):
+        assert path.endswith(".py")
+        self.path = path
+        self.visit(ast.parse(read_file(path)))
+
+    def summary(self):
+        lines = []
+        for path, node in self.found:
+            location = "{}:{}:{}".format(path, node.lineno, node.col_offset)
+            attr = node.value.id + "." + node.attr
+            lines.append("{} | {}".format(location, attr))
+        return "\n".join(lines)
+
+
+def test_finder(tmpdir):
+    src = """
+np.object
+numpy.int
+np.float(1.0)
+"""
+    path = tmpdir.join("test.py").strpath
+    with open(path, "w") as f:
+        f.write(src)
+
+    finder = NumpyBuiltinAliasFinder()
+    finder.find(path)
+
+    assert len(finder.found) == 3
+    assert finder.found[0][1].value.id == "np"
+    assert finder.found[0][1].attr == "object"
+    assert finder.found[1][1].value.id == "numpy"
+    assert finder.found[1][1].attr == "int"
+    assert finder.found[2][1].value.id == "np"
+    assert finder.found[2][1].attr == "float"
+
+
+def test_no_numpy_builtin_type_aliases():
+    finder = NumpyBuiltinAliasFinder()
+    for direcory in ["mlflow", "tests"]:
+        for path in iterate_python_scripts(direcory):
+            finder.find(path)
+
+    msg = (
+        "NumPy's aliases for built-in types (e.g. np.int) have been deprecated. "
+        "Please replace them by following this instruction: "
+        "https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated\n"  # noqa
+    )
+    assert finder.found == [], msg + finder.summary()
```

Comment:
The result looks like:

https://github.com/mlflow/mlflow/pull/4098/checks?check_run_id=1885889164#step:4:1673


```
E       AssertionError: numpy's built-in type aliases (e.g. np.int) have been deprecated. Please replace them by following this instruction: https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated
E         mlflow/types/schema.py:16:15 | np.object
E         mlflow/types/schema.py:48:50 | np.object
E         mlflow/pyfunc/__init__.py:299:23 | np.object
E         mlflow/pyfunc/__init__.py:302:48 | np.object
E         mlflow/pyfunc/__init__.py:814:83 | np.int
E         mlflow/pyfunc/__init__.py:814:91 | np.long
E         mlflow/utils/proto_json_utils.py:64:26 | np.object
E         tests/types/test_schema.py:144:59 | np.object
E         tests/types/test_schema.py:146:66 | np.object
E         tests/types/test_schema.py:148:76 | np.object
E         tests/types/test_schema.py:150:66 | np.object
E         tests/types/test_schema.py:152:65 | np.object
E         tests/types/test_schema.py:160:59 | np.str
E         tests/types/test_schema.py:164:60 | np.bool
E         tests/tensorflow/test_tensorflow2_model_export.py:250:43 | np.float
E         tests/tensorflow/test_tensorflow2_model_export.py:251:65 | np.float
E         tests/pyfunc/test_scoring_server.py:437:61 | np.object
E         tests/pyfunc/test_scoring_server.py:440:38 | np.bool
E         tests/pyfunc/test_spark.py:22:19 | np.int
E         tests/pyfunc/test_spark.py:22:27 | np.str
E         tests/pyfunc/test_spark.py:96:29 | np.int
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:150:14 | np.object
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:263:31 | np.object
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:264:31 | np.object
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:265:31 | np.object
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:266:31 | np.object
E         tests/pyfunc/test_model_export_with_loader_module_and_data_path.py:267:31 | np.object
E         tests/shap/test_shap.py:74:36 | np.float
E         tests/statsmodels/model_fixtures.py:156:80 | np.int
E         tests/statsmodels/model_fixtures.py:157:82 | np.int
E         tests/statsmodels/model_fixtures.py:158:82 | np.int
E         tests/onnx/test_onnx_model_export.py:168:35 | np.bool
E         tests/onnx/test_onnx_model_export.py:170:68 | np.bool
```